### PR TITLE
Arg patch

### PR DESCRIPTION
**Pull request by daveschaefer**
- Add flag to prompt for password (#6)
- Fix SSL usage (#7)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -154,14 +155,18 @@ func main() {
 			log.Fatalln("tls.LoadX509KeyPair:", err)
 		}
 		clientCert = append(clientCert, certs)
-		mysql.RegisterTLSConfig("custom", &tls.Config{
+		const customTLSName string = "customTLS"
```
Comment (razvanm) [inline main.go]: I think you can drop the `string`. The `custom` sounds better than `customTLS` to me. :P

File: main.go [hunk 2]
```diff
@@ -155,7 +160,32 @@ func main() {
...
+		dsnRegex := regexp.MustCompile(`(\w*):?\w*(@.+)`)
+		matches := dsnRegex.FindStringSubmatch(finalDsn)
+		if matches == nil {
+			fmt.Print("Incorrect format for dsn. Usage:\n")
+			flag.PrintDefaults()
+			os.Exit(1)
+		}
+
+		fmt.Print("Enter password: ")
+		// don't echo password to screen during input
```
Comment (razvanm) [inline main.go]: Please use uppercase for the first work and add a '.' at the end. Same below.

Reference: https://golang.org/doc/effective_go.html#commentary

File: main.go [hunk 3]
```diff
@@ -155,7 +160,32 @@ func main() {
...
+		bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
```
Comment (razvanm) [inline main.go]: Looking at [1] the `bytePassword` doesn't seem a suitable name. I think simply `password` would work fine.

[1] https://godoc.org/golang.org/x/crypto/ssh/terminal#Terminal.ReadPassword

File: main.go [hunk 4]
```diff
@@ -155,7 +160,32 @@ func main() {
 			ServerName:   *serverName,
 		})
 	}
-	db, err := sql.Open("mysql", *dsn)
+
+	var finalDsn = *dsn
+	if *prompt {
+		dsnRegex := regexp.MustCompile(`(\w*):?\w*(@.+)`)
```
Comment (razvanm) [inline main.go:177]: Can you add a comment explaining the regular expression? Experience taught me the hard way that even simple regular expression are easy to get wrong or misunderstand. :P

File: main.go (hunk 1 above)
Comment (daveschaefer) [inline main.go]: Sounds good :)

File: main.go (hunk 2 above)
Comment (daveschaefer) [inline main.go]: Sounds good

File: main.go (hunk 3 above)
Comment (daveschaefer) [inline main.go]: Sounds good

File: main.go (hunk 4 above)
Comment (daveschaefer) [inline main.go:177]: Yep, good idea :)

File: main.go [hunk 5]
```diff
@@ -168,6 +168,12 @@ func main() {
 	}
 
 	if *prompt {
+		// DSN strings look like:
+		// user:password@tcp(0.0.0.0:3306)/
+		// With this flag the user can avoid typing their password:
+		// user@tcp(0.0.0.0:3306)/
+		// Save text before : and after @ so we can insert the password
+		// to create a proper DSN string.
```
Comment (razvanm) [inline main.go:176]: A small reformatting suggestion:

```
        // DSN strings look like:
        //     user:password@tcp(0.0.0.0:3306)/
        // With this flag the user can avoid typing their password:
        //     user@tcp(0.0.0.0:3306)/
        // Save text before ':' and after '@' so we can insert the password
        // to create a proper DSN string.
```

Comment (daveschaefer) [inline main.go:176]: Sounds good! Done :)

Comment (razvanm) [inline main.go:176]: I apologize for being so picky but do you mind also adding the quotes I suggested? :-)

Comment (daveschaefer) [inline main.go:176]: Of course! No worries at all :) Sorry for missing it the first time

**Pull request merged**